### PR TITLE
MNT: List at least one microarch_level to avoid redundant builds

### PR DESCRIPTION
**Pull request by matthewfeickert**
Motivation: https://github.com/conda-forge/microarch-level-feedstock/issues/11

* Always list a `microarch_level` in `recipe/conda_build_config.yaml` to avoid `conda-build` searching for and setting an incorrect `microarch_level`, resulting in redundant builds.
   - c.f. https://github.com/conda-forge/conda-forge.github.io/pull/2197
* Bump build number.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-microarch_level:  # [x86_64]
-  - 1  # [x86_64]
+microarch_level:
+  - 1
   - 3  # [x86_64]
```
Comment (matthewfeickert) [inline recipe/conda_build_config.yaml:3]: I'm leaving this as

```yaml
- 3 # [x86_64]
```

and not the recommended

```yaml
- 3 # [unix and x86_64]
```

as I'm going to assume that you know what you're doing with your own feedstock. :+1:

**Comment by matthewfeickert** [comment]
@conda-forge/blasfeo this is ready for review.

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (traversaro) [reply recipe/conda_build_config.yaml:3]: Yes, this package enables the AVX instructions with explicit CMake options that trigger the use of internal assembly implementations, so even if AVX is not enabled via compilers flags it is still useful to enable it.

**Review by RSchwan — APPROVED** [review]

**Review by traversaro — APPROVED** [review]
Thanks a lot!

**Pull request merged**